### PR TITLE
workflows: fix already_harvested

### PR DESCRIPTION
**Pull request by jacquerie**
Closes #2150 

Implements properly the `already_harvested` task so that it filters
out arXiv records that are harvested on Legacy (closes #2214).

### CODE REVIEW

File: tests/integration/workflows/fixtures/oai_arxiv_record_with_plots.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@
             </author>
           </authors>
           <title>Spectra generated by a confined softcore Coulomb potential</title>
-          <categories>math-ph math.MP hep-ex</categories>
+          <categories>math-ph math.MP</categories>
```
Comment (jacquerie) [inline tests/integration/workflows/fixtures/oai_arxiv_record_with_plots.xml]: Ugly hack, will be fixed ASAP.

Comment (kaplun) [inline tests/integration/workflows/fixtures/oai_arxiv_record_with_plots.xml]: ? Uh? It's not a hack. It's that the `hep-ex` category should not be ingested (as of now), but this fixture had been created for a successful ingestion (unfortunately the wrong example of record had been picked up).

**Comment by kaplun** [comment]
I see you have added unit tests WRT my PR. Thanks. Anything more?

File: inspirehep/modules/workflows/tasks/matching.py [hunk 2]
```diff
@@ -219,12 +208,29 @@ def article_exists(obj, eng):
...
 
 
+def is_being_harvested_on_legacy(record):
+    """Return True if the record is being harvested on Legacy.
+
+    If the record belongs to one of the CORE arXiv categories then it
+    is already being harvested on Legacy.
+    """
+    arxiv_categories = list(chain.from_iterable(
+        get_value(record, 'arxiv_eprints.categories', [])))
```
Comment (kaplun) [inline inspirehep/modules/workflows/tasks/matching.py]: Mmh. I am not sure this is more readable than the previous implementation... but OK.

**Pull request merged**